### PR TITLE
Update profiler to detect updated minimum .NET versions

### LINKED ISSUE #1148: Profiler should detect unsupported .Net Core/Framework versions and detach.
### Feature Description
To avoid the major agent to break customer's applications, Profiler should detect unsupported .Net Core/Framework versions and detach. The unsupported .Net Core/Framework versions are:

- .Net Framework <= v4.6.1
- .Net Core <= 3.0

Refs: 

https://github.com/newrelic/newrelic-dotnet-agent/blob/da64b94e4f2fe6fd1b65dedd29051ee8704a10f3/src/Agent/NewRelic/Profiler/Profiler/CorProfilerCallbackImpl.h#L387

### PR DESCRIPTION
**Pull request by JcolemanNR**
## Description

Closes #1148 

Updates the profiler to detect newer profiler APIs for .NET Core 3.1, and .NET Framework 4.6.1. There was no new profiler interface introduced for .NET Framework 4.6.2, otherwise we would check for that.

# Author Checklist
- [x] Unit tests, Integration tests, and Unbounded tests completed
- [x] Performance testing completed with satisfactory results (if required)
- [ ] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated 

# Reviewer Checklist
- [x] Perform code review
- [x] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

### CODE REVIEW

**Comment by nr-ahemsath** [comment]
I would make sure to a) run the integration tests on Linux (Ubuntu) with the new profiler before merging this and b) smoke test the Linux profiler on Alpine since historically, changing how the profiler is built can cause Alpine problems.

File: .github/workflows/all_solutions.yml [hunk 1]
```diff
@@ -621,7 +616,6 @@ jobs:
             apt-get update -q -y && apt-get install -q -y curl
             export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1
             mkdir -p /usr/share/dotnet
-            curl -sSL https://dotnetcli.azureedge.net/dotnet/Sdk/2.1.818/dotnet-sdk-2.1.818-linux-arm64.tar.gz | tar -xzC /usr/share/dotnet
             curl -sSL https://dotnetcli.azureedge.net/dotnet/Sdk/3.1.414/dotnet-sdk-3.1.414-linux-arm64.tar.gz | tar -xzC /usr/share/dotnet
             curl -sSL https://dotnetcli.azureedge.net/dotnet/Sdk/5.0.401/dotnet-sdk-5.0.401-linux-arm64.tar.gz | tar -xzC /usr/share/dotnet
```
Comment (jaffinito) [inline .github/workflows/all_solutions.yml:620]: Do we also want to remove net5 since it is also EOL?

Comment (JcolemanNR) [reply .github/workflows/all_solutions.yml:620]: I don't think so, since we still target .NET 5 in some test apps, and aren't removing our support for it

**Comment by JcolemanNR** [comment]
> I would make sure to a) run the integration tests on Linux (Ubuntu) with the new profiler before merging this and b) smoke test the Linux profiler on Alpine since historically, changing how the profiler is built can cause Alpine problems.

@nr-ahemsath, I test integration tests on linux and enough passed to make me believe it is working. I also smoke tests on Alpine linux and the application starts up with the agent and accepts multiple requests without crashing.

Here is the sample dockerfile:
```
FROM mcr.microsoft.com/dotnet/aspnet:6.0-alpine3.15-amd64 AS base
WORKDIR /app
EXPOSE 80

FROM mcr.microsoft.com/dotnet/sdk:6.0-alpine3.15-amd64 AS build
WORKDIR /src
COPY ["TestDocker/TestDocker.csproj", "TestDocker/"]
RUN dotnet restore "TestDocker/TestDocker.csproj"
COPY . .
WORKDIR "/src/TestDocker"
RUN dotnet build "TestDocker.csproj" -c Release -o /app/build

FROM build AS publish
RUN dotnet publish "TestDocker.csproj" -c Release -o /app/publish

FROM base AS final
WORKDIR /usr/local
COPY ["newrelichome_x64_coreclr_linux/", "newrelichome_x64_coreclr_linux/"]

ENV CORECLR_ENABLE_PROFILING=1 \
CORECLR_PROFILER={36032161-FFC0-4B61-B559-F6C5D41BAE5A} \
CORECLR_NEWRELIC_HOME=/usr/local/newrelichome_x64_coreclr_linux \
CORECLR_PROFILER_PATH=/usr/local/newrelichome_x64_coreclr_linux/libNewRelicProfiler.so \
NEW_RELIC_LICENSE_KEY=YOUR_LICENSE_KEY \
NEW_RELIC_APP_NAME=YOUR_APP_NAME

WORKDIR /app
COPY --from=publish /app/publish .
ENTRYPOINT ["dotnet", "TestDocker.dll"]
```

**Comment by nr-ahemsath** [comment]
> > I would make sure to a) run the integration tests on Linux (Ubuntu) with the new profiler before merging this and b) smoke test the Linux profiler on Alpine since historically, changing how the profiler is built can cause Alpine problems.
> 
> @nr-ahemsath, I ran integration tests on linux and enough passed to make me believe it is working. I also smoke tested on Alpine linux and the application starts up with the agent and accepts multiple requests without crashing.

Thank you @JcolemanNR! 🥇

**Review by nr-ahemsath — APPROVED** [review]
Changes look good.  I pulled the branch and built it after doing a `git clean -dfx` and verified that, when attempting to profile a .NET Core 2.1 app, the profiler refused to attach. 👍

**Pull request merged**